src/client/opamClient.ml

```diff
@@ -1645,6 +1645,114 @@ let reinit ?(init_config=OpamInitDefaults.init_config()) ~interactive
   in
   OpamRepositoryState.drop rt
 
+let has_space s = OpamStd.String.contains_char s ' '
+
+let default_redirect_root = OpamFilename.Dir.of_string "C:\\opamroot"
+
+let setup_redirection target =
+  let {contents = {OpamStateConfig.original_root_dir = root; _}} =
+    OpamStateConfig.r
+  in
+  let target =
+    match target with
+    | Some target -> target
+    | None ->
+      OpamFilename.mkdir default_redirect_root;
+      let readme = OpamFilename.Op.(default_redirect_root // "ReadMe.txt") in
+      if not (OpamFilename.exists readme) then
+        OpamFilename.write readme
+          "This directory is used to contain redirected opam roots.\n\n\
+           The contents may be shared with other users on this system.";
+      OpamSystem.mk_unique_dir ~dir:(OpamFilename.Dir.to_string default_redirect_root) ()
```

Comment:
```suggestion
let default_redirect_root = lazy (OpamSystem.mk_unique_dir ~dir:"C:\\opamroot" ())

let setup_redirection target =
  let {contents = {OpamStateConfig.original_root_dir = root; _}} =
    OpamStateConfig.r
  in
  let target =
    match target with
    | Some target -> target
    | None ->
      OpamFilename.mkdir (Lazy.force default_redirect_root);
      let readme = OpamFilename.Op.(Lazy.force default_redirect_root // "ReadMe.txt") in
      if not (OpamFilename.exists readme) then
        OpamFilename.write readme
          "This directory is used to contain redirected opam roots.\n\n\
           The contents may be shared with other users on this system.";
```

Reply:
The idea is to `C:\opamroot\ReadMe.txt` explaining _all_ of the redirected files, not to have one in each directory